execution_workflows/TAPAS/bin/make_tapas_beds.py

```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env python3
+
+import sys
+tapa_quant_txt=open(sys.argv[1],'r')
+outbed_quant=open(sys.argv[2],'w')
+outbed_ident=open(sys.argv[3],'w')
+for ln in tapa_quant_txt:
+ ln=ln.strip().split('\t')
+ g_id,chrom,strand=ln[0].split('NM_')[1],ln[1],ln[2]
+ sites,scores=ln[3].split(','),ln[4].split(',')
+ if len(sites) == len(scores):
+  for i in range(len(sites)):
+   name=g_id+'_'+str(i)
+   score=scores[i]
+   if strand == '+':
+    chromStart=sites[i]
+    chromEnd=str(int(chromStart)+1)
+   elif strand == '-':
+    chromEnd=sites[i]
+    chromStart=str(int(chromEnd)-1)
```

Comment:
I need to double check this but this might introduce a shift of one 'to the left' for coordinates on the minus strand. Start coordinates are included in the range, ends are excluded. If sites[i] is the actual coordinate reported but it's set as the 'End' coordinate, it won't be considered as the single nucleotide (but one to the left will be). This is assuming that TAPAS's predictions are reported in 'zero based' convention (which I think I'd expect if you are pass TAPAS a BED as input

Reply:
saw that the input to tapas quant is tapas ref where the coordinates are taken from a bed file, so i think it's right that tapas output would be following the input bed file which is zero based. Does that mean we would treat both strands the same? In which we would have
chromStart=sites[i]
chromEnd=str(int(chromStart)+1)

I always get confused of coordinates and strands :,)

Reply:
I've dug into this a little deeper with some IGV browsing. I've concluded that the minus strand coordinates are shifted 1 to the left.

Example on the minus strand where TAPAS calls the distal site as the end of the reference transcript. With current code, the identification files (bottom two) are shifted one to the left of the final nucleotide of the annotated exon:

![tapas-one-off-error-from-reference-igv_snapshot](https://user-images.githubusercontent.com/49978382/192761394-402bf705-b6f8-4617-9b31-cac730c0a530.png)

Example on the plus strand (chr11:69,653,277-69,653,316) where TAPAS calls the distal site as the end of the reference transcript. Here can see identification files get the final nucleotide/PAS spot on (as suspected above):
![tapas-plus-strand-correct](https://user-images.githubusercontent.com/49978382/192762520-71862f49-0f3a-4af5-aa98-524d90e3252d.png)

@faricazjj is absolutely correct - the solution to correct the minus strand PAS coordinates is to set the reported coordinate as chromStart and +1 to get chromEnd. This means the procedure is identical regardless of the strand

Reply:
I have updated the coordinates with what [Farica suggested](https://github.com/iRNA-COSI/APAeval/pull/330#discussion_r972147372), where the + and - strands are treated the same as follows:
```
chromStart=sites[i]
chromEnd=str(int(chromStart)+1)
```